Sources/NIO/SelectorUring.swift

```diff
@@ -0,0 +1,359 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if SWIFTNIO_USE_IO_URING
+#if os(Linux) || os(Android)
+
+/// Represents the `poll` filters/events we might use from io_uring:
+///
+///  - `hangup` corresponds to `POLLHUP`
+///  - `readHangup` corresponds to `POLLRDHUP`
+///  - `input` corresponds to `POLLIN`
+///  - `output` corresponds to `POLLOUT`
+///  - `error` corresponds to `POLLERR`
+private struct UringFilterSet: OptionSet, Equatable {
+    typealias RawValue = UInt8
+
+    let rawValue: RawValue
+
+    static let _none = UringFilterSet([])
```

Comment:
Can I propose using `rawValue` directly here? Swift sometimes generates weird code around the array-literal constructions.

Reply:
Sure. Changing identical usage in epoll implementation too then.

Reply:
There was a reason for the epoll take on it:

```
/home/ubuntu/swiftnio/swift-nio/Sources/NIO/SelectorUring.swift:30:16: warning: static property '_none' produces an empty option set
    static let _none = UringFilterSet(rawValue: 0)
               ^
/home/ubuntu/swiftnio/swift-nio/Sources/NIO/SelectorUring.swift:30:16: note: use [] to silence this warning
    static let _none = UringFilterSet(rawValue: 0)
               ^                     ~~~~~~~~~~~~~
                                     ([])
```

Reply:
(so reverted to the [] pattern to squelch warnings)